ako-gateway-api/tests/utils.go

```diff
@@ -0,0 +1,357 @@
+		Name:     gatewayv1beta1.SectionName(name),
+		Port:     gatewayv1beta1.PortNumber(port),
+		Protocol: protocol,
+	}
+	if isTLS {
+		SetListenerTLS(&listner, gatewayv1beta1.TLSModeTerminate, "secret-example", "default")
+	}
+	gw.Spec.Listeners = append(gw.Spec.Listeners, listner)
+}
+
+func SetListenerTLS(l *gatewayv1beta1.Listener, tlsMode gatewayv1beta1.TLSModeType, secretName, secretNS string) {
+	l.TLS = &gatewayv1beta1.GatewayTLSConfig{Mode: &tlsMode}
+	namespace := gatewayv1beta1.Namespace(secretNS)
+	kind := gatewayv1beta1.Kind("Secret")
+	l.TLS.CertificateRefs = []gatewayv1beta1.SecretObjectReference{
+		{
+			Name:      gatewayv1beta1.ObjectName(secretName),
+			Namespace: &namespace,
+			Kind:      &kind,
+		},
+	}
+}
+func UnsetListenerTLS(l *gatewayv1beta1.Listener) {
+	l.TLS = &gatewayv1beta1.GatewayTLSConfig{}
+}
+
+func SetListenerHostname(l *gatewayv1beta1.Listener, hostname string) {
+	l.Hostname = (*gatewayv1beta1.Hostname)(&hostname)
+}
+func UnsetListenerHostname(l *gatewayv1beta1.Listener) {
+	var hname gatewayv1beta1.Hostname
+	l.Hostname = &hname
+}
+
+type TLSConfig struct {
+	Mode    string
+	Secrets []metav1.ObjectMeta
+}
+
+type Listener struct {
+	Name     string
+	Port     int32
+	Protocol string
+	Hostname string
+	TLS      TLSConfig
+}
+
+func GetListeners(ports []int32, secrets ...string) []Listener {
+	listeners := make([]Listener, 0, len(ports))
+	for _, port := range ports {
+		listener := Listener{
+			Name:     fmt.Sprintf("listener-%d", port),
+			Port:     port,
+			Protocol: "HTTPS",
+			Hostname: fmt.Sprintf("foo-%d.com", port),
+			TLS: TLSConfig{
+				Mode: "Terminate",
+			},
+		}
+		if len(secrets) > 0 {
+			listener.TLS.Secrets = make([]metav1.ObjectMeta, 0, len(secrets))
+			for _, secret := range secrets {
+				listener.TLS.Secrets = append(listener.TLS.Secrets,
+					metav1.ObjectMeta{
+						Name:      secret,
+						Namespace: "default",
+					},
+				)
+			}
+		}
+		listeners = append(listeners, listener)
+	}
+	return listeners
+}
+
+func GetListenerStatus(ports []int32, attachedRoutes []int32) []gatewayv1beta1.ListenerStatus {
+	listeners := make([]gatewayv1beta1.ListenerStatus, 0, len(ports))
+	for i, port := range ports {
+		listener := gatewayv1beta1.ListenerStatus{
+			Name:           gatewayv1beta1.SectionName(fmt.Sprintf("listener-%d", port)),
+			SupportedKinds: akogatewayapilib.SupportedKinds["HTTPS"],
+			AttachedRoutes: attachedRoutes[i],
+			Conditions: []metav1.Condition{
+				{
+					Type:               string(gatewayv1beta1.GatewayConditionAccepted),
+					Status:             metav1.ConditionTrue,
+					Message:            "Listener is valid",
+					ObservedGeneration: 1,
+					Reason:             string(gatewayv1beta1.GatewayReasonAccepted),
+				},
+			},
+		}
+		listeners = append(listeners, listener)
+	}
+	return listeners
+}
+
+type FakeGateway struct {
+	Name         string
+	Namespace    string
+	GatewayClass string
+	IPAddress    string
+	Listeners    []Listener
+}
```

Comment:
can we reuse the gateway struct from v1beta1

Reply:
I have made the changes as per the comment. Thanks